src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,1111 @@
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::BINARY, false);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.4054651081081644, 1, 1.4054651081081644, 0, 0 },
+    { 0, 1, 1.4054651081081644, 1.4054651081081644, 2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using sublinear
+ * as weighting scheme and smoothidf as true.
+ */
+BOOST_AUTO_TEST_CASE(SublinearSmoothIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::SUBLINEAR_TF, true);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.2876820724517808, 1.6931471805599454, 2.1802352704293200, 0, 0 },
+    { 0, 1.6931471805599454, 2.1802352704293200, 2.1802352704293200,
+      1.6931471805599454 },
+    { 1.2876820724517808, 1, 0, 1.2876820724517808, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using sublinear
+ * as weighting scheme and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(SublinearTfIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::SUBLINEAR_TF, false);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.4054651081081644, 1.6931471805599454, 2.3796592851687173, 0, 0 },
+    { 0, 1.6931471805599454, 2.3796592851687173, 2.3796592851687173,
+      2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using term
+ * Frequency as weighting scheme and smoothidf as true.
+ */
+BOOST_AUTO_TEST_CASE(TermFrequencySmoothIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::TERM_FREQUENCY, true);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 0.2575364144903562, 0.4, 0.5150728289807124, 0, 0 },
+    { 0, 0.2857142857142857, 0.3679091635576516, 0.3679091635576516,
+      0.2418781686514208 },
+    { 0.4292273574839269, 0.3333333333333333, 0, 0.4292273574839269, 0 }
+  };
```

Comment:
See the comment above.

Reply:
I agree with you, sklearn doesn't support term frequency mode, so I had to calculate it manually, for small input , the previous year case, and then got them corrected and then wrote to a CSV file for a bigger input.

Reply:
Ok